test-data-generator/src/main/java/com/adobe/platform/core/identity/services/datagenerator/AbstractDataGen.java

```diff
@@ -55,39 +56,58 @@ public synchronized void runDataGeneration(){
         }
     }
 
-    private synchronized void writeTestDataToCollections(Map<String, List<Document>> docs){
+    private synchronized void writeTestDataToCollections(Map<String, List<V4EncryptionCosmosDbClient.PojoizedJson>> docs){
         long docCount = docs.values().stream().map(a -> a.size()).mapToLong(Long::valueOf).sum();
         logger.info("Writing {} test documents to collections ...", docCount);
 
-        List<ResourceResponse<Document>> results = Observable.from(docs.entrySet())
-                .flatMap(entry -> Observable.from(entry.getValue()).map(a -> Pair.of(entry.getKey(), a)))
-                .flatMap(pair -> client.createDocument(pair.getKey(), pair.getValue()).onErrorResumeNext(Observable.empty()))
-                .toList()
-                .subscribeOn(Schedulers.computation())
-                .toBlocking()
-                .single();
+        List<CosmosItemResponse<V4EncryptionCosmosDbClient.PojoizedJson>> results = new ArrayList<>();
```

Comment:
we should have separate path for the existing code and encryption.

You should have one path for encryption and one path without and they should be separate. why are we passing everything through the encryption path now?

Reply:
i will have 2 client separate for each path , will send another iteration .

Reply:
added two clients